examples/ErrorHandling/HandleKeywordPolicyViolations.php

```diff
@@ -0,0 +1,263 @@
+                $exemptPolicyViolationKeys
+            );
+        } else {
+            print "No exemption request is sent because your keyword contained some non-exemptible"
+                . " policy violations" . PHP_EOL;
+        }
+    }
+
+    /**
+     * Collects all policy violation key that can be exempted for sending exemption request later.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @return PolicyViolationKey[] the exemptible policy violation keys
+     */
+    private static function fetchExemptPolicyViolationKeys(GoogleAdsException $googleAdsException)
+    {
+        $exemptPolicyViolationKeys = [];
+
+        printf("Google Ads failure details:%s", PHP_EOL);
+        foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+            /** @var GoogleAdsError $error */
+            printf(
+                "\t%s: %s%s",
+                $error->getErrorCode()->getErrorCode(),
+                $error->getMessage(),
+                PHP_EOL
+            );
+            if (!is_null($error->getDetails())
+                && !is_null($error->getDetails()->getPolicyViolationDetails())) {
+                $policyViolationDetails = $error->getDetails()->getPolicyViolationDetails();
+                printf("\tPolicy violation details:%s", PHP_EOL);
+                printf(
+                    "\t\tExternal policy name: '%s'%s",
+                    $policyViolationDetails->getExternalPolicyName(),
+                    PHP_EOL
+                );
+                printf(
+                    "\t\tExternal policy description: '%s'%s",
+                    $policyViolationDetails->getExternalPolicyDescription(),
+                    PHP_EOL
+                );
+                printf(
+                    "\t\tIs exemptible? '%s'%s",
+                    $policyViolationDetails->getIsExemptible() ? 'yes' : 'no',
+                    PHP_EOL
+                );
+
+                if ($policyViolationDetails->getIsExemptible() &&
+                    !is_null($policyViolationDetails->getKey())) {
+                    $policyViolationDetailsKey = $policyViolationDetails->getKey();
+                    $exemptPolicyViolationKeys[] = $policyViolationDetailsKey;
+                    printf("\t\tPolicy violation key:%s", PHP_EOL);
+                    printf(
+                        "\t\t\tName: '%s'%s",
+                        $policyViolationDetailsKey->getPolicyNameUnwrapped(),
+                        PHP_EOL
+                    );
+                    printf(
+                        "\t\t\tViolating text: '%s'%s",
+                        $policyViolationDetailsKey->getViolatingTextUnwrapped(),
+                        PHP_EOL
+                    );
+                }
+            }
+        }
+        return $exemptPolicyViolationKeys;
+    }
+
+    /**
+     * Sends exemption requests for creating a keyword.
+     *
+     * @param int $customerId
```

Comment:
The documentation of all parameters is missing.

Reply:
This is intentional because I don't have anything much to say. This is the same for the Java library case.